include/swift/AST/CASTBridging.h

```diff
@@ -331,42 +373,186 @@ void *UnresolvedDotExpr_create(BridgedASTContext cContext, void *base,
+                             BridgedSourceLoc cEqualsLoc,
+                             void *_Nullable opaqueRawValue);
+
+SWIFT_NAME(
+    "StructDecl_create(astContext:declContext:structKeywordLoc:name:nameLoc:"
+    "genericParamList:inheritedTypes:genericWhereClause:braceRange:)")
+BridgedDeclContextAndDecl StructDecl_create(
+    BridgedASTContext cContext, BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cStructKeywordLoc, BridgedIdentifier cName,
+    BridgedSourceLoc cNameLoc, void *_Nullable opaqueGenericParamList,
+    BridgedArrayRef cInheritedTypes, void *_Nullable opaqueGenericWhereClause,
+    BridgedSourceRange cBraceRange);
+
+SWIFT_NAME(
+    "ClassDecl_create(astContext:declContext:classKeywordLoc:name:nameLoc:"
+    "genericParamList:inheritedTypes:genericWhereClause:braceRange:isActor:)")
+BridgedDeclContextAndDecl ClassDecl_create(
+    BridgedASTContext cContext, BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cClassKeywordLoc, BridgedIdentifier cName,
+    BridgedSourceLoc cNameLoc, void *_Nullable opaqueGenericParamList,
+    BridgedArrayRef cInheritedTypes, void *_Nullable opaqueGenericWhereClause,
+    BridgedSourceRange cBraceRange, _Bool isActor);
+
+SWIFT_NAME("ProtocolDecl_create(astContext:declContext:protocolKeywordLoc:name:"
+           "nameLoc:primaryAssociatedTypeNames:inheritedTypes:"
+           "genericWhereClause:braceRange:)")
+BridgedDeclContextAndDecl ProtocolDecl_create(
+    BridgedASTContext cContext, BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cProtocolKeywordLoc, BridgedIdentifier cName,
+    BridgedSourceLoc cNameLoc, BridgedArrayRef cPrimaryAssociatedTypeNames,
+    BridgedArrayRef cInheritedTypes, void *_Nullable opaqueGenericWhereClause,
+    BridgedSourceRange cBraceRange);
+
+SWIFT_NAME(
+    "AssociatedTypeDecl_create(astContext:declContext:associatedtypeKeywordLoc:"
+    "name:nameLoc:inheritedTypes:defaultType:genericWhereClause:)")
+void *AssociatedTypeDecl_create(BridgedASTContext cContext,
+                                BridgedDeclContext cDeclContext,
+                                BridgedSourceLoc cAssociatedtypeKeywordLoc,
+                                BridgedIdentifier cName,
+                                BridgedSourceLoc cNameLoc,
+                                BridgedArrayRef cInheritedTypes,
+                                void *_Nullable opaqueDefaultType,
+                                void *_Nullable opaqueGenericWhereClause);
+
+SWIFT_NAME("ExtensionDecl_create(astContext:declContext:extensionKeywordLoc:"
+           "extendedType:inheritedTypes:genericWhereClause:braceRange:)")
+BridgedDeclContextAndDecl ExtensionDecl_create(
+    BridgedASTContext cContext, BridgedDeclContext cDeclContext,
+    BridgedSourceLoc cExtensionKeywordLoc, void *opaqueExtendedType,
+    BridgedArrayRef cInheritedTypes, void *_Nullable opaqueGenericWhereClause,
+    BridgedSourceRange cBraceRange);
+
+typedef enum ENUM_EXTENSIBILITY_ATTR(open) {
```

Comment:
I simply followed the existing pattern really. `closed` exports the enum as `@frozen`. `open` *probably* exists as a requisite for importing C enums that are not `closed` into a Swift module compiled with library evolution enabled, to avoid accidental openness. If binary compatibility is not a concern in the foreseeable future, we can indeed use `closed` or just remove the attribute altogether.

Reply:
Ok [I looked this up](https://github.com/apple/swift/blob/f5407533ee8477f5bee381a0c4a95ac6ac92b410/docs/HowSwiftImportsCAPIs.md?plain=1#L946). We need the attribute or else the enum will be imported as a struct. I will mark them as closed then. As for `BridgedImportKind`, I suggest keeping it in sync with `ImportKindSpecifierOptions` (open).

Reply:
Thanks 👍🏽